great_expectations/core/usage_statistics/execution_environment.py

```diff
@@ -0,0 +1,125 @@
+
+This module contains utilities for identifying installed package dependencies
+to help enable the core team to safely upgrade package dependencies to gain
+access to features of new package versions.
+
+    Typical usage example:
+        ge_execution_environment: GEExecutionEnvironment = GEExecutionEnvironment()
+        dependencies: List[PackageInfo] = ge_execution_environment.dependencies
+
+"""
+
+import enum
+from dataclasses import dataclass
+from importlib import metadata
+from typing import List, Optional
+
+from packaging import version
+
+from great_expectations.core.usage_statistics.package_dependencies import GEDependencies
+
+
+class InstallEnvironment(enum.Enum):
+    DEV = "dev"
+    REQUIRED = "required"
+
+
+@dataclass
+class PackageInfo:
+    package_name: str
+    installed: bool
+    install_environment: InstallEnvironment
+    version: Optional[version.Version]
+
+
+class GEExecutionEnvironment:
+    """The list of GE dependencies with version and install information.
+
+    This does not return any dependencies that are not specified directly in
+    either requirements.txt or any requirements-dev*.txt files.
+
+    Attributes: None
+    """
+
+    def __init__(self):
+        self._ge_dependencies: GEDependencies = GEDependencies()
+        self._all_installed_packages = None
+        self._get_all_installed_packages()
+
+        self._dependencies = []
+        self._build_required_dependencies()
+        self._build_dev_dependencies()
+
+    @property
+    def dependencies(self) -> List[PackageInfo]:
+        """The list of GE dependencies with version and install information.
+
+        This does not return any dependencies that are not specified directly in
+        either requirements.txt or any requirements-dev*.txt files.
+
+        Attributes: None
+        """
+        return self._dependencies
+
+    def _build_required_dependencies(self) -> None:
+        dependency_list: List[PackageInfo] = self._build_dependencies_info(
+            self._ge_dependencies.get_required_dependency_names(),
+            install_environment=InstallEnvironment.REQUIRED,
+        )
+        self._dependencies.extend(dependency_list)
+
+    def _build_dev_dependencies(self) -> None:
+        dependency_list: List[PackageInfo] = self._build_dependencies_info(
+            self._ge_dependencies.get_dev_dependency_names(),
+            install_environment=InstallEnvironment.DEV,
+        )
+        self._dependencies.extend(dependency_list)
+
+    def _get_all_installed_packages(self) -> List[str]:
+        """Get all installed packages. Used to check if dependency is installed or not.
+
+        Returns:
+            list of string names of packages that are installed
+        """
+        if self._all_installed_packages is None:
+            # Only retrieve once
+            self._all_installed_packages = [
+                item.metadata.get("Name") for item in metadata.distributions()
+            ]
```

Comment:
Is "Name" guaranteed? If not, should we perhaps filter out nulls before returning?

Reply:
It is from the output of `metadata.distributions()` but better safe than sorry, added.